components/charts/ElectricityCapacityChart.tsx

```diff
@@ -79,21 +73,11 @@ export default function ElectricityCapacityChart({
   // Calculate linear trend line if requested
   let trendValues: number[] = [];
   if (showTrend && capacityValues.length > 1) {
-    // Calculate linear regression
-    const n = capacityValues.length;
-    const sumX = labels.reduce((sum: number, _: string, index: number) => sum + index, 0);
-    const sumY = capacityValues.reduce((sum: number, val: number) => sum + val, 0);
-    const sumXY = capacityValues.reduce((sum: number, val: number, index: number) => sum + (index * val), 0);
-    const sumX2 = labels.reduce((sum: number, _: string, index: number) => sum + (index * index), 0);
-
-    const slope = (n * sumXY - sumX * sumY) / (n * sumX2 - sumX * sumX);
-    const intercept = (sumY - slope * sumX) / n;
-
-    trendValues = labels.map((_: string, index: number) => intercept + slope * index);
```

Comment:
In this previous implementation, label positioning is also used to calculate the linear regression. This seems unnecessary since labels and data positioning should have identical lengths. In the new calculation, I only used data values to compute the linear regression values

See components/charts/utils/trendCalculator.ts